types/evidence.go

```diff
@@ -179,8 +180,61 @@ func EvidenceToProto(evidence Evidence) (*tmproto.Evidence, error) {
 		tp := &tmproto.Evidence{
 			Sum: &tmproto.Evidence_PotentialAmnesiaEvidence{
 				PotentialAmnesiaEvidence: &tmproto.PotentialAmnesiaEvidence{
-					VoteA: voteA,
-					VoteB: voteB,
+					VoteA:       voteA,
+					VoteB:       voteB,
+					HeightStamp: evi.HeightStamp,
+				},
+			},
+		}
+		return tp, nil
+
+	case AmnesiaEvidence:
+		ev, err := EvidenceToProto(evi.PotentialAmnesiaEvidence)
+		if err != nil {
+			return nil, err
+		}
+		// EvidenceToProto turns the evidence into Evidence oneof but we neeed a specific type
+		paepb := ev.GetPotentialAmnesiaEvidence()
+		if paepb == nil {
+			return nil, errors.New("provided evidence is not potential amnesia evidence")
+		}
+
+		polc, err := evi.Polc.ToProto()
+		if err != nil {
+			return nil, err
+		}
+
+		tp := &tmproto.Evidence{
+			Sum: &tmproto.Evidence_AmnesiaEvidence{
+				AmnesiaEvidence: &tmproto.AmnesiaEvidence{
+					PotentialAmnesiaEvidence: paepb,
+					Polc:                     polc,
+				},
+			},
+		}
+
+		return tp, nil
+	case *AmnesiaEvidence:
+		ev, err := EvidenceToProto(evi.PotentialAmnesiaEvidence)
```

Comment:
can we extract this into a func and call for both `AmnesiaEvidence` and `*AmnesiaEvidence` ?

Reply:
I can find out if it's possible - but we do this for all evidences (copy them twice for each instance)